Sources/PostgresNIO/PostgresDatabase+Query.swift

```diff
@@ -74,7 +74,16 @@ public struct PostgresQueryMetadata {
             self.command = .init(parts[0])
             self.oid = Int(parts[1])
             self.rows = Int(parts[2])
-        case "DELETE", "UPDATE", "SELECT", "MOVE", "FETCH", "COPY":
+        case "SELECT":
+            // AWS Redshift does not return the actual row count as defined in the postgres wire spec:
+            // https://www.postgresql.org/docs/13/protocol-message-formats.html in section `CommandComplete`
+            guard parts.count == 1 || parts.count == 2 else {
+                return nil
+            }
+            self.command = .init(parts[0])
+            self.oid = nil
+            self.rows = parts.count == 2 ? Int(parts[1]) : 0
+        case "DELETE", "UPDATE", "MOVE", "FETCH", "COPY":
```

Comment:
Small nit. Let’s make this a special case that we handle.

```suggestion
        case "SELECT" where parts.count == 1:
            // AWS Redshift does not return the actual row count as defined in the postgres wire spec for SELECT:
            // https://www.postgresql.org/docs/13/protocol-message-formats.html in section `CommandComplete`
            self.command = "SELECT"
            self.oid = nil
            self.rows = nil
        case "SELECT", "DELETE", "UPDATE", "MOVE", "FETCH", "COPY":
```

Reply:
Thats much better! Thanks